api/src/upload-documents-in-folder.js

```diff
@@ -0,0 +1,29 @@
+const fs = require('fs');
+const path = require('path');
+
+const db = require('./db');
+const settingService = require('./services/settings');
+
+const uploadDocumentsInFolder = async folderPath => {
+  const parsedFiles = fs.readdirSync(folderPath).map(filename => {
+    const filePath = path.resolve(folderPath, filename);
+    const fileContent = fs.readFileSync(filePath).toString();
+    const parsed = JSON.parse(fileContent);
+    
+    return {
+      name: filename,
+      doc: parsed,
+    };
+  });
+  
+  // Settings doc requires special casing because it does not have an id
+  const settingsDoc = parsedFiles.find(file => file.name === 'settings');
+  if (settingsDoc) {
+    await settingService.update(settingsDoc.doc);
+  }
+
+  const notSettingsDocs = parsedFiles.filter(file => !settingsDoc || file !== settingsDoc).map(file => file.doc);
+  await db.medic.bulkDocs(notSettingsDocs);
```

Comment:
`bulkDocs` has an annoying property that it doesn't throw if any of the uploads fail (for example, due to conflict). I think it's important that if they fail we block starting API because it could lead to difficult to detect errors. To do this you have to iterate over the results and check the `ok` property, eg: https://github.com/medic/medic/blob/master/api/src/services/messaging.js#L276

We have to do this a lot - maybe it's worth pulling out a little service or extending the db module with an extra method or param which causes the promise to reject if any upload fails. You could also build in a retry option if you like but that may be overkill for now...

Reply:
I'm counting around 8-10 places which might benefit from a library like this. I pushed the quick solution and will revisit with a PouchDB plugin as available.

Reply:
With this as the blocker for 3.7, mind if I open a new issue for this?

Reply:
Absolutely split off a separate issue for it!